packages/tabs/src/index.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Flex, Text } from 'pcln-design-system'
+import styled from 'styled-components'
+
+const TabTitle = styled(Text)`
+  border-bottom: ${props =>
+    props.isActive ? `4px solid ${props.activeColor}` : 'none'};
+  cursor: pointer;
+`
+
+class Tabs extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      activeTab: 0
+    }
+  }
+
+  render() {
+    const { activeColor, defaultColor, tabs } = this.props
+    const children = React.Children.toArray(this.props.children)
+
+    return (
+      <React.Fragment>
+        <Flex bg="lightGray" px={1} pt={2}>
+          {tabs.map((tab, i) => (
+            <TabTitle
+              color={this.state.activeTab === i ? activeColor : defaultColor}
+              key={i}
+              mx={3}
+              py={2}
+              onClick={() => this.setState({ activeTab: i })}
```

Comment:
Binding functions in the render is bad practice, I see why you did it in this situation but my recommendation would be to make a separate `TabTitle` which could use `memo`. Binding in a stateless components render, especially a memoized one, is much better as it will rerender much less often.

Good explanation of memo: https://scotch.io/tutorials/react-166-reactmemo-for-functional-components-rendering-control

Reply:
Just noticed below that you are using enzyme for testing, memo will not work with the current version of enzyme but people are working on a fix https://github.com/airbnb/enzyme/pull/1914 I would recommend just making a stateless component for now and adding memo to it later, that is if people think this overall approach is a good one

Reply:
Be sure to see the [official docs on `React.memo`](https://reactjs.org/docs/react-api.html#reactmemo) and the [Optimizing Performance](https://reactjs.org/docs/optimizing-performance.html) page